File: uva/10551.py
# 10551 - Basic Remains

import sys

def to_base_10(n, b):

	r, p = 0, 1
	while n != 0:
		r += (n % 10) * p
		p *= b
		n //= 10

	return r

def to_base(n, b):
	
	r = ''
	while n >= b:
		r = chr(ord('0') + n % b) + r
		n //= b

	return chr(ord('0') + n) + r

for line in sys.stdin:
	linput = list(map(int, line.split()))
	if linput[0] == 0:
		break;
	
	b, p, m = linput

	if b == 10:
		print(p % m)
	else:
		p = to_base_10(p, b)
		m = to_base_10(m, b)
		print(to_base(p % m, b))

File: uva/10551.cpp
// 10551 - Basic Remains

#include <cstdio>

int to_base(int n, const int & b_in, const int & b_out)
{
	int r = 0;
	int p = 1;
	
	while(n)
	{
		r += (n % b_out) * p;
		p *= b_in;
		n /= b_out;
	}

	return r;
}

int main()
{
	int b, m;
	char p[1001];
	long long r;

	while(scanf("%d", &b) && b)
	{
		scanf("%s %d", p, &m);
		m = to_base(m, b, 10);
		
		r = 0;
		for(int i = 0; p[i]; i++)
		{
			r = (r * b) % m;
			r = (r + p[i] - '0') % m;
		}
		
		printf("%d\n", to_base(r, 10, b));
	}

	return 0;
}

